core/libp2p/protocol/identify/utils.hpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_ID_UTILS_HPP
+#define KAGOME_ID_UTILS_HPP
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <tuple>
+
+#include "libp2p/connection/stream.hpp"
+#include "libp2p/host.hpp"
+#include "libp2p/network/connection_manager.hpp"
+#include "libp2p/peer/protocol.hpp"
+
+namespace libp2p::protocol::detail {
+  /**
+   * Get a tuple of stringified <PeerId, Multiaddress> of the peer the (\param
+   * stream) is connected to
+   */
+  std::tuple<std::string, std::string> getPeerIdentity(
+      const std::shared_ptr<libp2p::connection::Stream> &stream);
+
+  /**
+   * Get collection of peers, to which we have at least one active connection
+   * @param host of this node
+   * @param conn_manager of this node
+   * @return PeerInfo-s
+   */
+  std::vector<peer::PeerInfo> getActivePeers(
+      Host &host, network::ConnectionManager &conn_manager);
+
+  /**
+   * Open a stream to each peer this host is connected to and execute a provided
+   * function
+   * @param host of this node
+   * @param conn_manager of this node
+   * @param protocol to open stream on
+   * @param handler to be executed
+   */
+  void streamToEachConnectedPeer(Host &host,
```

Comment:
I think this can be useful in other places. This actually looks like a broadcast. 
Maybe put TODO with a task `Make a component out of this`?

Reply:
Yeah, for some reason I removed a ToDo, which was there